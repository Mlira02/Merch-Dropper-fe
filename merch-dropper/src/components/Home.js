import React, { useState, useEffect } from "react";
import { useAuth0 } from "./Auth/Auth";

import "../App.css";
import {
  Button,
  Row,
  Col,
  Card,
  CardImg,
  CardText,
  CardBody,
  CardTitle,
  CardSubtitle,
  Jumbotron,
  Container,
  Media
} from "reactstrap";
import ProductDisplay from "./ProductDisplay";

const Home = ({ history }) => {
  return (
    <div className="homepage" style={{ backgroundColor: "#F3F3F3" }}>
      {/* Begin hero image */}
      <div className="jumboParent">
        <Jumbotron className="jumboCont">
          <Container>
            <Row>
              <Col className="column" sm={{ size: "auto" }}>
                <h1
                  className="display-3 customh1"
                  style={{
                    textAlign: "center",
                    color: "white",
                    fontWeight: "bold"
                  }}
                >
                  CUSTOM
                </h1>
                <h2
                  className="display-4 tshirth1"
                  style={{
                    textAlign: "center",
                    color: "white",
                    fontWeight: "bold"
                  }}
                >
                  T-SHIRTS
                </h2>
                <h1
                  className="andMore"
                  style={{
                    textAlign: "center",
                    color: "white",
                    fontWeight: "bold"
                  }}
                >
                  AND MORE
                </h1>
                <div className="line2" />
                <p
                  className="p2"
                  style={{
                    textAlign: "center",
                    color: "white",
                    fontWeight: "bold"
                  }}
                >
                  Need merch? We've got your back.
                </p>
                <p
                  className="p2"
                  style={{
                    textAlign: "center",
                    color: "white",
                    margin: "5% 0"
                  }}
                >
                  Full service merchandising solutions from Merch Dropper and
                  Scalable Press, no matter the size.
                </p>
                <p style={{ textAlign: "center" }}>
                  <Button
                    color="primary"
                    className="letsGo"
                    href="/designshirt"
                  >
                    Let's Go!
                  </Button>
                  <br />
                  <Button
                    outline
                    color="light"
                    className="learnMore"
                    onClick={() => {
                      history.push("/learnmore");
                    }}
                  >
                    Learn More
                  </Button>
                </p>
              </Col>
              <Col
                className="customImg"
                sm={{ size: "auto" }}
                style={{ width: "25vw", height: "60vh", margin: "0 5vw" }}
              >
                <Media
                  object
                  src="https://i.ibb.co/wQWhj2C/Group-6.png"
                  style={{ width: "15vw" }}
                />
                <Media
                  object
                  src="https://i.ibb.co/7XQ70x1/Group-5.png"
                  style={{ width: "15vw" }}
                />
              </Col>
            </Row>
          </Container>
        </Jumbotron>
      </div>
      {/* End hero image */}
      {/* Begin featured shops section */}
      <div style={{ width: "100vw" }} className="featuredDiv">
        <Container
          style={{
            padding: "0 5%",
            borderTop: "2px solid #026FC2",
            borderLeft: "2px solid #026FC2",
            borderTopLeftRadius: "10px"
          }}
        >
          <Row>
            <h1
              className="mb-4 featuredText"
              style={{
                color: "#026FC2",
                backgroundColor: "#F3F3F3",
                marginTop: "-2%",
                width: "25%",
                textAlign: "center"
              }}
            >
              Featured Shops
            </h1>
          </Row>
          <Row style={{ width: "100%" }}>
            <Media
              object
              src="https://i.ibb.co/HTvq2CY/image-6.png"
              style={{ width: "30%", height: "10%" }}
            />
            <Media
              object
              src="https://i.ibb.co/XzjYthv/image-7.png"
              style={{ width: "30%", height: "10%", margin: "0 5%" }}
            />
            <Media
              object
              src="https://i.ibb.co/vzsn9kf/image-9.png"
              style={{ width: "30%", height: "10%" }}
            />
          </Row>
        </Container>
        <div>
          <Row
            style={{
              width: "95vw",
              display: "flex",
              justifyContent: "space-evenly",
              margin: "auto"
            }}
            className="mt-5 cards"
          >
            <Card className="cardCont">
              <CardImg
                top
                width="100%"
                src="https://i.ibb.co/zhpXpNy/Bg.png"
                alt="Card image cap"
                style={{ height: "70%" }}
              />
              <CardBody>
                <CardTitle className="cardTitle">Kitten Mug</CardTitle>
                <CardText className="cardText">
                  Like cats? We make merch your way.
                </CardText>
              </CardBody>
            </Card>

            <Card className="cardCont">
              <CardImg
                top
                width="100%"
                src="https://i.ibb.co/jH9JXZy/Image-2.png"
                alt="Card image cap"
                style={{ height: "70%" }}
              />
              <CardBody>
                <CardTitle className="cardTitle" style={{ color: "#026FC2" }}>
                  iPhone 11 Case
                </CardTitle>
                <CardText className="cardText">
                  Like cats? We make merch your way.
                </CardText>
              </CardBody>
            </Card>

            <Card className="cardCont">
              <CardImg
                top
                width="100%"
                src="https://i.ibb.co/6bnHXp2/Bg-1.png"
                alt="Card image cap"
                style={{ height: "70%" }}
              />
              <CardBody>
                <CardTitle className="cardTitle" style={{ color: "#026FC2" }}>
                  Adorable Jumper
                </CardTitle>
                <CardText className="cardText">
                  Like cats? We make merch your way.
                </CardText>
              </CardBody>
            </Card>

            <Card className="cardCont">
              <CardImg
                top
                width="100%"
                src="https://i.ibb.co/N1TNcJg/Screen-Shot-2020-02-05-at-11-15-1.png"
                alt="Card image cap"
                style={{ height: "70%" }}
              />
              <CardBody>
                <CardTitle className="cardTitle" style={{ color: "#026FC2" }}>
                  Fitted Tee
                </CardTitle>
                <CardText className="cardText">
                  Like cats? We make merch your way.
                </CardText>
              </CardBody>
            </Card>
          </Row>
        </div>
      </div>
      {/* End featured shops section */}
    </div>
  );
};

export default Home;
