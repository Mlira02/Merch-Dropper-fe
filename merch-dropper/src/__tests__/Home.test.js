import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from '../components/Home';

it('renders without crashing', () => {
  render(<Home />);
});

it('renders Home correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});