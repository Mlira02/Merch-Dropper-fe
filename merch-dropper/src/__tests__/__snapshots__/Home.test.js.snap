// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home correctly 1`] = `
<div
  className="homepage"
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <div
    className="jumboParent"
  >
    <div
      className="jumboCont jumbotron"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="column col-sm-auto"
          >
            <h1
              className="display-3 customh1"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              CUSTOM
            </h1>
            <h2
              className="display-4 tshirth1"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              T-SHIRTS
            </h2>
            <h1
              className="andMore"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              AND MORE
            </h1>
            <div
              className="line2"
            />
            <p
              className="p2"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Need merch? We've got your back.
            </p>
            <p
              className="p2"
              style={
                Object {
                  "color": "white",
                  "margin": "5% 0",
                  "textAlign": "center",
                }
              }
            >
              Full service merchandising solutions from Merch Dropper and Scalable Press, no matter the size.
            </p>
            <p
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <a
                aria-label={null}
                className="letsGo btn btn-primary"
                href="/designshirt"
                onClick={[Function]}
              >
                Let's Go!
              </a>
              <br />
              <button
                aria-label={null}
                className="learnMore btn btn-outline-light"
                onClick={[Function]}
                type="button"
              >
                Learn More
              </button>
            </p>
          </div>
          <div
            className="customImg col-sm-auto"
            style={
              Object {
                "height": "60vh",
                "margin": "0 5vw",
                "width": "25vw",
              }
            }
          >
            <img
              className="media-object"
              src="https://i.ibb.co/wQWhj2C/Group-6.png"
              style={
                Object {
                  "width": "15vw",
                }
              }
            />
            <img
              className="media-object"
              src="https://i.ibb.co/7XQ70x1/Group-5.png"
              style={
                Object {
                  "width": "15vw",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="featuredDiv"
    style={
      Object {
        "width": "100vw",
      }
    }
  >
    <div
      className="container"
      style={
        Object {
          "borderLeft": "2px solid #026FC2",
          "borderTop": "2px solid #026FC2",
          "borderTopLeftRadius": "10px",
          "padding": "0 5%",
        }
      }
    >
      <div
        className="row"
      >
        <h1
          className="mb-4 featuredText"
          style={
            Object {
              "backgroundColor": "#F3F3F3",
              "color": "#026FC2",
              "marginTop": "-2%",
              "textAlign": "center",
              "width": "25%",
            }
          }
        >
          Featured Shops
        </h1>
      </div>
      <div
        className="row"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <img
          className="media-object"
          src="https://i.ibb.co/HTvq2CY/image-6.png"
          style={
            Object {
              "height": "10%",
              "width": "30%",
            }
          }
        />
        <img
          className="media-object"
          src="https://i.ibb.co/XzjYthv/image-7.png"
          style={
            Object {
              "height": "10%",
              "margin": "0 5%",
              "width": "30%",
            }
          }
        />
        <img
          className="media-object"
          src="https://i.ibb.co/vzsn9kf/image-9.png"
          style={
            Object {
              "height": "10%",
              "width": "30%",
            }
          }
        />
      </div>
    </div>
    <div>
      <div
        className="mt-5 cards row"
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-evenly",
            "margin": "auto",
            "width": "95vw",
          }
        }
      >
        <div
          className="cardCont card"
        >
          <img
            alt="Card image cap"
            className="card-img-top"
            src="https://i.ibb.co/zhpXpNy/Bg.png"
            style={
              Object {
                "height": "70%",
              }
            }
            width="100%"
          />
          <div
            className="card-body"
          >
            <div
              className="cardTitle card-title"
            >
              Kitten Mug
            </div>
            <p
              className="cardText card-text"
            >
              Like cats? We make merch your way.
            </p>
          </div>
        </div>
        <div
          className="cardCont card"
        >
          <img
            alt="Card image cap"
            className="card-img-top"
            src="https://i.ibb.co/jH9JXZy/Image-2.png"
            style={
              Object {
                "height": "70%",
              }
            }
            width="100%"
          />
          <div
            className="card-body"
          >
            <div
              className="cardTitle card-title"
              style={
                Object {
                  "color": "#026FC2",
                }
              }
            >
              iPhone 11 Case
            </div>
            <p
              className="cardText card-text"
            >
              Like cats? We make merch your way.
            </p>
          </div>
        </div>
        <div
          className="cardCont card"
        >
          <img
            alt="Card image cap"
            className="card-img-top"
            src="https://i.ibb.co/6bnHXp2/Bg-1.png"
            style={
              Object {
                "height": "70%",
              }
            }
            width="100%"
          />
          <div
            className="card-body"
          >
            <div
              className="cardTitle card-title"
              style={
                Object {
                  "color": "#026FC2",
                }
              }
            >
              Adorable Jumper
            </div>
            <p
              className="cardText card-text"
            >
              Like cats? We make merch your way.
            </p>
          </div>
        </div>
        <div
          className="cardCont card"
        >
          <img
            alt="Card image cap"
            className="card-img-top"
            src="https://i.ibb.co/N1TNcJg/Screen-Shot-2020-02-05-at-11-15-1.png"
            style={
              Object {
                "height": "70%",
              }
            }
            width="100%"
          />
          <div
            className="card-body"
          >
            <div
              className="cardTitle card-title"
              style={
                Object {
                  "color": "#026FC2",
                }
              }
            >
              Fitted Tee
            </div>
            <p
              className="cardText card-text"
            >
              Like cats? We make merch your way.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
